boards/feather-m0-wifi/Kconfig

```diff
@@ -10,14 +10,7 @@ config BOARD
 config BOARD_FEATHER_M0_WIFI
     bool
     default y
-    select CPU_MODEL_SAMD21G18A
-    select HAS_PERIPH_ADC
-    select HAS_PERIPH_I2C
-    select HAS_PERIPH_PWM
-    select HAS_PERIPH_RTC
-    select HAS_PERIPH_RTT
-    select HAS_PERIPH_SPI
-    select HAS_PERIPH_TIMER
-    select HAS_PERIPH_UART
-    select HAS_PERIPH_USBDEV
-    select HAS_HIGHLEVEL_STDIO
+    select BOARD_FEATHER_M0
```

Comment:
I'd preferably split common feather_m0 features into its own symbol in `boards/feather-m0/Kconfig` and select that instead. The `BOARD_` symbols are usually used to identify which board is used and to set the default to `BOARD`. Although in this case `BOARD` should get the proper value because of the order of inclusion I find it cleaner if we want to factor this out.

Reply:
Should we split the board Kconfig from this PR?